src/main/java/com/synopsys/integration/alert/provider/blackduck/actions/BlackDuckDistributionTestAction.java

```diff
@@ -78,22 +79,22 @@ public MessageResult testConfig(String configId, FieldModel fieldModel, FieldAcc
 
                 BlackDuckApiTokenValidator blackDuckAPITokenValidator = new BlackDuckApiTokenValidator(blackDuckProperties);
                 if (!blackDuckAPITokenValidator.isApiTokenValid()) {
-                    throw AlertFieldException.singleFieldError(ProviderDescriptor.KEY_PROVIDER_CONFIG_NAME, "User permission failed, cannot read notifications from Black Duck.");
+                    fieldStatuses.add(AlertFieldStatus.error(ProviderDescriptor.KEY_PROVIDER_CONFIG_NAME, "User permission failed, cannot read notifications from Black Duck."));
                 }
             }
         } else {
-            throw AlertFieldException.singleFieldError(ProviderDescriptor.KEY_PROVIDER_CONFIG_NAME, "A provider configuration is required");
+            fieldStatuses.add(AlertFieldStatus.error(ProviderDescriptor.KEY_PROVIDER_CONFIG_NAME, "A provider configuration is required"));
         }
-
-        return new MessageResult("Successfully tested BlackDuck provider fields");
+        return new MessageResult("Successfully tested BlackDuck provider fields", fieldStatuses);
```

Comment:
This could be confusing to handle further down the line, since the message will say success but there could be errors